cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java

```diff
@@ -0,0 +1,218 @@
+
+  private static final ExtensionContext.Namespace NAMESPACE =
+      ExtensionContext.Namespace.create(CassandraExtension.class);
+
+  private static final String KEY = "cassandra";
+
+  @Override public boolean supportsParameter(final ParameterContext parameterContext,
+      final ExtensionContext extensionContext) throws ParameterResolutionException {
+    final Class<?> type = parameterContext.getParameter().getType();
+    return Session.class.isAssignableFrom(type) || Cluster.class.isAssignableFrom(type);
+  }
+
+  @Override public Object resolveParameter(final ParameterContext parameterContext,
+      final ExtensionContext extensionContext) throws ParameterResolutionException {
+
+    Class<?> type = parameterContext.getParameter().getType();
+    if (Session.class.isAssignableFrom(type)) {
+      return getOrCreate(extensionContext).session;
+    } else if (Cluster.class.isAssignableFrom(type)) {
+      return getOrCreate(extensionContext).cluster;
+    }
+
+    throw new ExtensionConfigurationException(
+        String.format(Locale.ROOT, "%s supports only %s or %s but yours was %s",
+        CassandraExtension.class.getSimpleName(),
+        Session.class.getName(), Cluster.class.getName(), type.getName()));
+  }
+
+  static ImmutableMap<String, String> getDataset(String resourcePath) {
+    return ImmutableMap.of("model",
+        Sources.of(CassandraExtension.class.getResource(resourcePath))
+            .file().getAbsolutePath());
+  }
+
+  /**
+   * Register cassandra resource in root context so it can be shared with other tests
+   */
+  private static CassandraResource getOrCreate(ExtensionContext context) {
+    // same cassandra instance should be shared across all extension instances
+    return context.getRoot()
+        .getStore(NAMESPACE)
+        .getOrComputeIfAbsent(KEY, key -> new CassandraResource(), CassandraResource.class);
+  }
+
+  /**
+   * Whether to run this test.
+   * <p>Enabled by default, unless explicitly disabled
+   * from command line ({@code -Dcalcite.test.cassandra=false}) or running on incompatible JDK
+   * version (see below).
+   *
+   * <p>As of this wiring Cassandra 4.x is not yet released and we're using 3.x
+   * (which fails on JDK11+). All cassandra tests will be skipped if
+   * running on JDK11+.
+   *
+   * @see <a href="https://issues.apache.org/jira/browse/CASSANDRA-9608">CASSANDRA-9608</a>
+   * @return {@code true} if test is compatible with current environment,
+   *         {@code false} otherwise
+   */
+  @Override public ConditionEvaluationResult evaluateExecutionCondition(
+      final ExtensionContext context) {
+    boolean enabled = CalciteSystemProperty.TEST_CASSANDRA.value();
+    Bug.upgrade("remove JDK version check once current adapter supports Cassandra 4.x");
+    boolean compatibleJdk = TestUtil.getJavaMajorVersion() < 11;
+    if (enabled && compatibleJdk) {
+      return ConditionEvaluationResult.enabled("Cassandra enabled");
```

Comment:
Is this still valid?
Does it probably support Java 11?

Reply:
Cassandra support for JDK11 was introduced in 4.x (see [CASSANDRA-9608](https://issues.apache.org/jira/browse/CASSANDRA-9608) we still use 3.6